### PR TITLE
Sandbox Pattern

### PR DESCRIPTION
**Pull request by oonosan**
Added Sanbox abstract class with loadUser() method

### CODE REVIEW

File: src/DonateTo.WebApplication/ClientApp/src/app/app.component.ts [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
+export class AppComponent {
+  title = 'app';
+
+  param = { value: 'donate' };
+
+  constructor(translate: TranslateService, public appSandbox: AppSandbox) {
+    this.appSandbox.setupLanguage();
+
+    // this language will be used as a fallback when a translation isn't found in the current language
+    translate.setDefaultLang('es');
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/app.component.ts:19]: The default lang should be english.

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/store/actions/settings.action.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Action } from '@ngrx/store';
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/store/actions/settings.action.ts:1]: Please create a folder settings-->actions

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/store/index.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { createSelector } from 'reselect';
+
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/store/index.ts:2]: this ts should go inside the settings folder

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/store/reducers/settings.reducer.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import '@ngrx/core/add/operator/select';
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/store/reducers/settings.reducer.ts:1]: this reducer should go inside settings folder

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/sandbox/base.sandbox.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { Store } from '@ngrx/store';
+import * as store from '../store';
+
+export abstract class Sandbox {
+  public culture$ = this.appState$.select(store.getSelectedCulture);
+  public culture: string;
+
+  constructor(protected appState$: Store<store.State>) {}
+
+  public loadUser(): void {}
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/sandbox/base.sandbox.ts:10]: Please add a TODO comment which says "Implement this after the issue 57 is merged"

File: src/DonateTo.WebApplication/ClientApp/src/app/app.sandbox.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { Sandbox } from './shared/sandbox/base.sandbox';
+import { Store } from '@ngrx/store';
+import * as store from './shared/store/settings';
+import * as settingsActions from './shared/store/settings/settings.action';
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/app.sandbox.ts:5]: This import setttingsActions should be imported with the line 4

File: src/DonateTo.WebApplication/ClientApp/src/app/home/home.sandbox.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+import { Store } from '@ngrx/store';
+import { Subscription } from 'rxjs';
+import { Sandbox } from '../shared/sandbox/base.sandbox';
+import * as store from '../shared/store';
+
+@Injectable()
+export class HomeSandbox extends Sandbox {
+  public culture$ = this.appState$.select(store.getSelectedCulture);
+  public language$ = this.appState$.select(store.getSelectedLanguage);
+  public availableLanguages$ = this.appState$.select(store.getAvailableLanguages);
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/home/home.sandbox.ts:11]: Remove these lines. They are inherited from the abstract class Sandbox

File: src/DonateTo.WebApplication/ClientApp/src/app/home/home.sandbox.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/home/home.sandbox.ts:12]: Add these variables: 
- **samples$** this has to retrieve the data from the store using appState$.select(store.getAllSamples);
- **samplesLoading$** this has to retrieve the data from the store using appState$.select(store.getSamplesLoading);

File: src/DonateTo.WebApplication/ClientApp/src/app/home/home.sandbox.ts [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+  /**
+   * Loads sample from the server
+   */
+  public createSample(): void {}
+
+  /**
+   * Loads samples from the server
+   */
+  public loadSamples(): void {}
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/home/home.sandbox.ts:28]: This method should dispatch the action LoadSamples
`this.appState$.dispatch(new LoadSamples())`

**Pull request merged**